Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/PhysicsPicking.cpp

```diff
@@ -0,0 +1,189 @@
+                    continue;
+                }
+
+                const AZ::Vector3 parentPos = pose->GetWorldSpaceTransform(parentIndex).m_position;
+                const AZ::Vector3 bonePos = pose->GetWorldSpaceTransform(jointIndex).m_position;
+
+                const AZ::Vector3 boneDir = (parentPos - bonePos).GetNormalized();
+                const float boneLength = (parentPos - bonePos).GetLength();
+                float t1 = AZ::Constants::FloatMax;
+                float t2 = AZ::Constants::FloatMax;
+                int numIntersections =
+                    AZ::Intersect::IntersectRayCappedCylinder(rayOrigin, rayDirection, bonePos, boneDir, boneLength, PickingMargin, t1, t2);
+                if (numIntersections > 0)
+                {
+                    float distance = (numIntersections == 1) ? t1 : AZ::GetMin(t1, t2);
+                    CompareIntersection(closestIntersection, distance, parentIndex);
+                }
+            }
+        }
+
+        if (AZ::RHI::CheckBitsAny(m_renderFlags, EMotionFX::ActorRenderFlags::RagdollColliders))
+        {
+            Physics::CharacterColliderConfiguration* ragdollColliderConfiguration =
+                actorInstance->GetActor()->GetPhysicsSetup()->GetColliderConfigByType(EMotionFX::PhysicsSetup::Ragdoll);
+            for (const Physics::CharacterColliderNodeConfiguration& nodeConfig : ragdollColliderConfiguration->m_nodes)
+            {
+                const EMotionFX::Actor* actor = actorInstance->GetActor();
+                const EMotionFX::Node* joint = actor->GetSkeleton()->FindNodeByName(nodeConfig.m_name.c_str());
+                if (joint)
+                {
+                    const size_t jointIndex = joint->GetNodeIndex();
+                    const EMotionFX::Transform& actorInstanceGlobalTransform = actorInstance->GetWorldSpaceTransform();
+                    const EMotionFX::Transform& emfxNodeGlobalTransform =
+                        actorInstance->GetTransformData()->GetCurrentPose()->GetModelSpaceTransform(jointIndex);
+                    const AZ::Transform worldTransform = (emfxNodeGlobalTransform * actorInstanceGlobalTransform).ToAZTransform();
+                    for (const AzPhysics::ShapeColliderPair& shapeColliderPair : nodeConfig.m_shapes)
+                    {
+                        const AZ::Transform colliderOffsetTransform = AZ::Transform::CreateFromQuaternionAndTranslation(
+                            shapeColliderPair.first->m_rotation, shapeColliderPair.first->m_position);
+                        const AZ::Transform colliderGlobalTransform = worldTransform * colliderOffsetTransform;
+                        const AZ::TypeId colliderType = shapeColliderPair.second->RTTI_GetType();
+                        if (colliderType == azrtti_typeid<Physics::SphereShapeConfiguration>())
+                        {
+                            auto* sphere = static_cast<Physics::SphereShapeConfiguration*>(shapeColliderPair.second.get());
```

Comment:
Is this the same code (actually more then the selected code area) that we also use for rendering the colliders? Any chance we can share any of that?

Reply:
There is some duplication, but it's different enough that I don't think it would help that much to try to deduplicate it.